File: narrative-lifecycle-dashboard-handoff-v0.1/scripts/deep_probe.py
import sys
import requests
from bs4 import BeautifulSoup
import json
import os

def fetch_url(url):
    try:
        headers = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7)'}
        resp = requests.get(url, headers=headers, timeout=10)
        resp.raise_for_status()
        
        # Check if PDF
        if 'application/pdf' in resp.headers.get('Content-Type', ''):
            return {"type": "pdf", "content": "PDF parsing requires pdfplumber/PyMuPDF (mocked for now, assuming PDF downloaded)", "url": url}
            
        soup = BeautifulSoup(resp.text, 'html.parser')
        
        # Remove script/style
        for script in soup(["script", "style"]):
            script.decompose()
            
        text = soup.get_text(separator=' ', strip=True)
        return {"type": "html", "title": soup.title.string if soup.title else "", "text": text[:5000], "url": url} # limit to 5000 chars for context
        
    except Exception as e:
        return {"error": str(e), "url": url}

if __name__ == "__main__":
    url = sys.argv[1]
    print(json.dumps(fetch_url(url), ensure_ascii=False))

File: narrative-lifecycle-dashboard-handoff-v0.1/scripts/deep_probe.ts
import * as fs from 'fs';
import * as path from 'path';

async function fetchUrl(url: string) {
  try {
    const response = await fetch(url, {
      headers: {
        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36',
        'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8'
      }
    });

    if (!response.ok) {
      throw new Error(`HTTP Error: ${response.status}`);
    }

    const contentType = response.headers.get('content-type') || '';
    if (contentType.includes('application/pdf')) {
      return { type: 'pdf', content: 'PDF fetched. Requires PDF parser.', url };
    }

    const text = await response.text();
    const titleMatch = text.match(/<title[^>]*>([^<]+)<\/title>/i);
    const title = titleMatch ? titleMatch[1].trim() : '';

    const bodyMatch = text.match(/<body[^>]*>([\s\S]*)<\/body>/i);
    let body = bodyMatch ? bodyMatch[1] : text;
    body = body.replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, ' ');
    body = body.replace(/<style\b[^<]*(?:(?!<\/style>)<[^<]*)*<\/style>/gi, ' ');
    body = body.replace(/<[^>]+>/g, ' ');
    body = body.replace(/\s+/g, ' ').trim();

    return { type: 'html', title, text: body.substring(0, 8000), url };

  } catch (error: any) {
    return { error: error.message, url };
  }
}

async function main() {
  const url = process.argv[2];
  if (!url) {
    console.error("Usage: npx tsx scripts/deep_probe.ts <url>");
    process.exit(1);
  }
  const result = await fetchUrl(url);
  console.log(JSON.stringify(result, null, 2));
}

main();
